### DISCUSSION TITLE
I can't figure out how to decrease the analyzed fps in a live rtsp stream using detect.py

### DISCUSSION THREAD

**Discussion by superalino** [ultralytics/yolov5#7725] [Q&A]
As the title suggests, I am trying to use detect.py to analyze 6 live rtsp streams (ipcams). Unfortunately I don't have enough computing power and I would like to decrease the analyzed fps.

Is there an easy way to do that?

**Comment by glenn-jocher** [comment]
@superalino you can customize the streamloader here to suite your needs.
https://github.com/ultralytics/yolov5/blob/d059d1da03aee9a3c0059895aa4c7c14b7f25a9e/utils/datasets.py#L302-L389


Note that multi-stream support is automatic in YOLOv5, you just place your addresses in a streams.txt and pass the textfile to --source

**Reply by superalino** [reply]
unfortunately I can't understand the mechanism

I thought the part to change was this:


def update(self, i, cap, stream): 
         # Read stream `i` frames in daemon thread 
         n, f, **read** = 0, self.frames[i], 1  # frame number, frame array, inference every 'read' frame 
         while cap.isOpened() and n < f: 
             n += 1 
             # _, self.imgs[index] = cap.read() 
             cap.grab() 
             **if n % read == 0:**  #  it is an operation divided by 1 and the remainder is checked
                 success, im = cap.retrieve() 
                 if success: 
                     self.imgs[i] = im 
                 else: 
                     LOGGER.warning('WARNING: Video stream unresponsive, please check your IP camera connection.') 
                     self.imgs[i] = np.zeros_like(self.imgs[i]) 
                     cap.open(stream)  # re-open stream if signal was lost 
             time.sleep(1 / self.fps[i])  # wait time 

So I increased the read var to 10 and I expected that the inference was on 1/10 frames

1 % 10 == 0 (false) -> skip
2 % 10 == 0 (false) -> skip
...
10 % 10 == 0 (true) -> get frame
...
19 % 10 == 0 (false) -> skip
20 % 10 == 0 (true) -> get frame
...
and so on

but it did not work.

what's the difference between cap.retrieve, cap.grab and cap.read ?

**Reply by glenn-jocher** [reply]
@superalino see cv2 documentation for cap methods

**Reply by superalino** [reply]
ok:

cap.grab = Grabs the next frame from video file or capturing device.
cap.retrieve = Decodes and returns the grabbed video frame
cap.read = Grabs, decodes and returns the next video frame.

and I'm stuck. I'll try harder to understand.

**Comment by Kabuc0** [comment]
@superalino hello, i am currently also working on object detection with a stream (http or rtsp) as input. Unfortunately I have about 3 seconds delay. Do you have the same problem?

**Comment by glenn-jocher** [comment]
Good news 😃! Your original issue may now be fixed ✅ in PR #8243. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Reply by Kabuc0** [reply]
@glenn-jocher YES, its working now without any delay!!! Appreciate your fast work really really much. Thank you!!!!!!!
